lib/utils/net.go

```diff
@@ -41,6 +41,11 @@ func ClientAddrContext(ctx context.Context, src net.Addr, dst net.Addr) context.
 	return context.WithValue(ctx, ClientDstAddrContextKey, dst)
 }
 
+// ClientSrcAddrContext sets client source address.
+func ClientSrcAddrContext(ctx context.Context, src net.Addr) context.Context {
```

Comment:
Can't we use `ClientAddrContext` since it keeps compatibility with existing code?

Reply:
We can use the existing `ClientAddrContext` but have to call `ClientAddrFromContext` first to get the `dst` and use `new-src` and `dst` with `ClientAddrContext`. I would prefer not to go around like this.